.github/workflows/add-changelog-snippet.yml

```
@@ -0,0 +1,63 @@
+name: Add changelog.d snippet
+
+on:
+  pull_request_target:
+    # "labeled" is also triggered when a PR is created with one or more labels
+    types: [synchronize, labeled]
+
+# Prevent the workflow from running multiple jobs at once when a PR is created
+# with multiple labels:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  add:
+    runs-on: ubuntu-latest
+    if: >
+      github.repository == 'datalad/datalad'
+        && contains(github.event.pull_request.labels.*.name, 'CHANGELOG-missing')
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+
+      - name: Configure Git identity
+        run: |
+          git config --global user.name "github-actions"
+          git config --global user.email "github-actions@users.noreply.github.com"
```

```suggestion
          git config --global user.name "DataLad Bot"
          git config --global user.email "bot@datalad.org"
```

as we already use in `.autorc`.